search_handler.py

```diff
@@ -0,0 +1,62 @@
+from pdb import set_trace
+import json
+import urllib
+from lxml import etree
+
+from core.config import Configuration as CoreConfiguration
+from config import config as search_handler_config
+
+class Configuration(CoreConfiguration):
+    @classmethod
+    def load(cls):
+        cls.instance = search_handler_config
+        return cls.instance
+
+config = Configuration.load()
+
+
+from core.opensearch import OpenSearchDocument
+from core.util.opds_writer import AtomFeed, OPDSFeed
+from core.external_search import ExternalSearchIndex
+
+def search_handler(event, context):
+    search_url = ""
+    input = event.get('input')
+    if input:
+        request_headers = input.get('Headers')
+        if request_headers:
+            protocol = request_headers.get('X-Forwarded-Proto', '')
+            host = request_headers.get('Host', '')
+            path = input.get('path', '')
+            search_url = protocol + '://' + host + path
+
+    query = event.get('queryStringParameters').get("q")
```

Comment:
Yes. The input format when you use Lambda via API gateway is documented here: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-set-up-simple-proxy.html#api-gateway-simple-proxy-for-lambda-input-format